### PR TITLE
Update ScalaCheck to 1.14.3

### PR DESCRIPTION
**Pull request by ashawley**
- Drop the sourced copy of ScalaCheck.
- Add back a dependency for the latest version.
- Removed vestigial occurrences of `overrideParameters` since it's good to use different initial seeds.  The one in `mutable.BitSetProperties` was causing failures for a `subset` property test.

### CODE REVIEW

**Comment by ashawley** [comment]
Dropping a sourced library like this should typically wait until after the 2.14.x branch was cut, to avoid bootstrapping dependencies during 2.14 milestones, but it sounds like the former won't be occurring.

**Review by dwijnand — APPROVED** [review]
LGTM, thanks!

**Comment by lrytz** [comment]
The intellij project definition should be also removed

  - https://github.com/scala/scala/blob/2.13.x/src/intellij/scala.ipr.SAMPLE#L187
  - https://github.com/scala/scala/blob/2.13.x/src/intellij/scala.ipr.SAMPLE#L406-L412
  - https://github.com/scala/scala/blob/2.13.x/src/intellij/scalacheck-src.iml.SAMPLE

**Comment by dwijnand** [comment]
> The intellij project definition should be also removed

See https://github.com/scala/scala/blob/2.13.x/src/intellij/README.md#updating-the-sample-files

**Comment by ashawley** [comment]
I'm not familiar with changing the Intellij configuration here, but I gave it a try.

File: build.sbt [hunk 1]
```diff
@@ -749,23 +740,15 @@ lazy val junit = project.in(file("test") / "junit")
...
   .settings(skip in publish := true)
   .settings(
-    // Enable forking to workaround https://github.com/sbt/sbt/issues/4009. We also need a lazy Framework
-    // written in Java to further reduce ClassLoader problems when bootstrapping:
+    // Enable forking to workaround https://github.com/sbt/sbt/issues/4009.
     fork in Test := true,
-    testFrameworks := Seq(TestFramework("org.scalacheck.LazyFramework")),
-    // When we switch to sbt 1.3.0 it should be possible to set the ClassLoaderLayeringStrategy instead:
-    //classLoaderLayeringStrategy in Test := ClassLoaderLayeringStrategy.Flat,
-    // customise framework for early access to https://github.com/rickynils/scalacheck/pull/388
```
Comment (SethTisue) [inline build.sbt:767]: @eed3si9n here on the 2.13.x branch, we're on sbt 1.3 now. should we revisit this? should the comment stay in place?

File: build.sbt [hunk 2]
```diff
@@ -749,23 +740,15 @@ lazy val junit = project.in(file("test") / "junit")
...
-    // TODO remove this when we upgrade scalacheck
-    //testFrameworks := Seq(TestFramework("org.scalacheck.CustomScalaCheckFramework")),
     javaOptions in Test += "-Xss1M",
-    //Make scalacheck print full stack traces
-    testOptions in Test += Tests.Argument(TestFramework("org.scalacheck.LazyFramework"), "-verbosity", "2"),
```
Comment (SethTisue) [inline build.sbt:772]: why removed? (the full-stack trace part, I mean)

File: build.sbt (hunk 1 above)
Comment (eed3si9n) [reply build.sbt:767]: In https://github.com/sbt/sbt/issues/4009#issuecomment-462314260 @szeiger said:

> Switching to 1.3.0-bin-20190203T202003 does not make a difference by itself. The framework still gets initialized with the wrong classloader. It does work when I switch to `ClassLoaderLayeringStrategy.Flat`. 

So if the whole point of forking was https://github.com/sbt/sbt/issues/4009, and the point of `org.scalacheck.LazyFramework` was a workaround for https://github.com/typelevel/scalacheck/issues/185, then we should remove `fork in Test := true` and replace it with:

```scala
Test / classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.Flat,
```

Comment (SethTisue) [reply build.sbt:767]: @ashawley care to give that a try as part of this PR? (if not, bringing the comment up to date would also be acceptable)

Comment (ashawley) [reply build.sbt:767]: Sure, I'll give it a try.  I just misunderstood the comment.  

Enabling that new setting and disabling forking causes trouble for me, locally.  Let's see what Travis and Jenkins say, though.

File: build.sbt (hunk 2 above)
Comment (ashawley) [reply build.sbt:772]: I don't think it's necessary, but I can put it back.

File: build.sbt (hunk 1 above)
Comment (ashawley) [reply build.sbt:767]: Unless I missed something, it seems it's a strange failure unrelated to the previous tickets:

```
[error] 1 of 17 test tasks failed:
[error] - scalacheck/test
[error]   - scalacheck/test:executeTests failed: java.lang.ExceptionInInitializerError
[error] testAll failed due to previous errors
[error] (testAll) testAll failed due to previous errors
```
Locally, the sbt log says:
```
> last
[debug] > Exec(scalacheck/test, Some(796486cf-838a-4e65-84f5-bd03999e7788), Some(CommandSource(console0)))
[debug] Evaluating tasks: checkBuildSources
[debug] Running task... Cancel: Signal, check cycles: false, forcegc: true
[debug] Evaluating tasks: scalacheck / Test / testOnly
[debug] Running task... Cancel: Signal, check cycles: false, forcegc: true
[warn] javaOptions will be ignored, fork is set to false
[error] java.lang.ExceptionInInitializerError
[error] 	at java.lang.Class.forName0(Native Method)
[error] 	at java.lang.Class.forName(Class.java:348)
[error] 	at org.scalacheck.Platform$.loadModule(Platform.scala:69)
[error] 	at org.scalacheck.ScalaCheckRunner$BaseTask.<init>(ScalaCheckFramework.scala:55)
[error] 	at org.scalacheck.ScalaCheckRunner$$anon$2.<init>(ScalaCheckFramework.scala:93)
[error] 	at org.scalacheck.ScalaCheckRunner.rootTask(ScalaCheckFramework.scala:93)
[error] 	at org.scalacheck.ScalaCheckRunner.$anonfun$tasks$2(ScalaCheckFramework.scala:44)
[error] 	at org.scalacheck.ScalaCheckRunner$$Lambda$4658/977034192.apply(Unknown Source)
[error] 	at scala.collection.ArrayOps$.map$extension(ArrayOps.scala:924)
[error] 	at org.scalacheck.ScalaCheckRunner.tasks(ScalaCheckFramework.scala:44)
[error] 	at sbt.TestRunner.tasks(TestFramework.scala:107)
```

Comment (ashawley) [reply build.sbt:767]: I'll just bring back the comment as Seth suggested.

**Pull request merged**